### DISCUSSION TITLE
How URL matches

### DISCUSSION THREAD

**Discussion by Dhruv-Tech-Apps** [Dhruv-Techapps/acf-docs#348] [General]
# URL Matching System

The extension offers a single `Auto start` configuration alongside the flexibility of multiple `Manual start` configurations that can match specific URLs.

## Determining Configuration Priority

Consider the webpage URL: `https://test.getautoclicker.com/?username=Dhruv`

### 🥈 Partial Match / Regular Expression Match
- Matches with the webpage URL.
- In case of multiple matches with partial or regular expression criteria, the first match is selected.
- Examples:
  - `https://test.getautoclicker.com/`
  - `https://test.getautoclicker.com/?username=`
  - `https://.+getautoclicker\.com/.+`

### 🥇 Exact Match (Top Priority)
- Requires an exact match of the URL with the webpage URL.
- Examples:
  - `https://test.getautoclicker.com/?username=Dhruv`

> Users can effortlessly trigger manual configurations using their assigned hotkeys, providing a convenient way for repeated utilization against the same URL.

---

## Auto Configuration
- Find single configuration based on above match criteria and start to execute

## Manual Configuration
- Matches all configuration based on above match criteria and map against hot key
- Use hot key to start the configuration

**Comment by ghost** [comment]
How do I get the extension to click a link that comes up after a poll? I have it set to click the correct choice and then the submit button. Once the submit button is clicked, the current results pop up and there is a new link that says "return to poll" which is a javascript link. I can easily just have the extension refresh the page, but I would rather have this link clicked. I have tried everything but can't figure it out.

//*[@id="PDI_answer61283458"]
//*[@id="pd-vote-button13729341"]
//*[@id="PDI_container13729341"]/div/div/div/div/div[6]/div/span[1]/a

That third line is what the XPath gives me for the link, but I don't understand it. I have also tried the class action to no avail.

This is the page: https://co.milesplit.com/articles/347682/colorado-girls-performer-of-the-week-april-29-may-4

**Reply by dharmesh-hemaram** [reply]
This link you shared is showing 404 no content. Can you also open log and check what error you getting there.

You can also use `//a[contains(text(),"link text goes here")]`

**Reply by ghost** [reply]
That is odd on the 404, I am not seeing that.
I don't see an error on the page, the thing in the corner just says done, but the like is not clicked. I tried the code you provided and still no dice.
When I look at the console, it shows that refused to run javascrpit URL error. So, I will try the legacy version and see if that does anything.

**Reply by ghost** [reply]
Legacy version worked. Thanks!

**Comment by tonymetzgerjr** [comment]
Hi

Lookking for assistance with having a wild card in the Config file URL.  I'm using Auto-Clicker to click on a Download button from a website using a URL similar to: https://stockjunk.com/stocks/nnnn/financials/  where nnnn is a stock symbol that will change with each company that I am studying.  Can I use the asterix (*) in place of nnnn and have the config file to all?  Thanks

**Comment by Dhruv-Tech-Apps** [comment]
Try .*Sent from my iPhoneOn 19 Nov 2024, at 3:02 AM, tonymetzgerjr ***@***.***> wrote:﻿
Hi
Lookking for assistance with having a wild card in the Config file URL.  I'm using Auto-Clicker to click on a Download button from a website using a URL similar to: https://stockjunk.com/stocks/nnnn/financials/  where nnnn is a stock symbol that will change with each company that I am studying.  Can I use the asterix (*) in place of nnnn and have the config file to all?  Thanks

**Reply by tonymetzgerjr** [reply]
It seems to be working fine.  Many thanks!
